src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/AbstractEntityAIHerder.java

```diff
@@ -0,0 +1,432 @@
+package com.minecolonies.coremod.entity.ai.citizen.herders;
+
+import com.minecolonies.api.util.InventoryUtils;
+import com.minecolonies.api.util.constant.ToolType;
+import com.minecolonies.coremod.colony.Structures;
+import com.minecolonies.coremod.colony.jobs.AbstractJob;
+import com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIInteract;
+import com.minecolonies.coremod.entity.ai.util.AIState;
+import com.minecolonies.coremod.entity.ai.util.AITarget;
+import com.minecolonies.coremod.util.StructureWrapper;
+import net.minecraft.entity.passive.EntityAnimal;
+import net.minecraft.init.Items;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.Mirror;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraftforge.items.wrapper.InvWrapper;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.minecolonies.api.util.constant.ToolLevelConstants.TOOL_LEVEL_WOOD_OR_GOLD;
+import static com.minecolonies.coremod.entity.ai.util.AIState.*;
+
+/**
+ * Abstract class for all Citizen Herder AIs
+ */
+public abstract class AbstractEntityAIHerder<J extends AbstractJob, T extends EntityAnimal> extends AbstractEntityAIInteract<J>
+{
+
+    /**
+     * Amount of animals needed to bread.
+     */
+    public static final int NUM_OF_ANIMALS_TO_BREED = 2;
+
+    /**
+     * Butchering attack damage.
+     */
+    private static final int BUTCHERING_ATTACK_DAMAGE = 5;
+
+    /**
+     * Distance two animals need to be inside to breed.
+     */
+    private static final int DISTANCE_TO_BREED = 10;
+
+    /**
+     * Amount of animals to keep per Hut Level.
+     */
+    private int maxAnimalMultiplier;
+
+    /**
+     * Delays used to setDelay()
+     */
+    private static final int DELAY_TWENTY = 20;
+    private static final int DELAY_FOURTY = 40;
+    private static final int DELAY_ONE_HUNDRED = 100;
+
+    /**
+     * Number of actions needed to dump inventory.
+     */
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this assignment of "maxAnimalMultiplier". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3010)